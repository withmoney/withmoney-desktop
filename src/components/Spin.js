import React from 'react';
import PropTypes from 'prop-types';

const Spin = ({ white }) => {
  const color = white ? 'white' : 'black';

  return (
    <img src={`/static/img/spin-${color}.svg`} alt="spin" className="spin" />
  );
};

Spin.propTypes = {
  white: PropTypes.bool,
};

Spin.defaultProps = {
  white: false,
};

export default Spin;
