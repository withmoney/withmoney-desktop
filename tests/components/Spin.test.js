import React from 'react';
import renderer from 'react-test-renderer';
import Spin from '../../src/components/Spin';

const setup = Component => (props = {}) => <Component {...props} />;

describe('Spin', () => {
  let Component;

  beforeAll(() => {
    Component = setup(Spin);
  });

  it('should render spin black', () => {
    const wrapper = renderer.create(Component());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render spin white', () => {
    const wrapper = renderer.create(Component({ white: true }));

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
