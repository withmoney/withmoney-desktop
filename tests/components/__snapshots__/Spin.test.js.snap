// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin should render spin black 1`] = `
<img
  alt="spin"
  className="spin"
  src="/static/img/spin-black.svg"
/>
`;

exports[`Spin should render spin white 1`] = `
<img
  alt="spin"
  className="spin"
  src="/static/img/spin-white.svg"
/>
`;
